app/controllers/OtherReasonController.scala

```diff
@@ -128,14 +128,11 @@ class OtherReasonController @Inject()(whenDidBecomeUnablePage: WhenDidBecomeUnab
         previousUploadsState <- uploadJourneyRepository.getUploadsForJourney(request.session.get(SessionKeys.journeyId))
       } yield {
         if (request.cookies.get("jsenabled").isEmpty || featureSwitching.isEnabled(NonJSRouting)) {
-          mode match {
-            case NormalMode => Redirect(controllers.routes.OtherReasonController.onPageLoadForFirstFileUpload(mode))
-            case CheckMode => if (previousUploadsState.exists(_.count(_.fileStatus == READY) > 0)) {
-              Redirect(controllers.routes.OtherReasonController.onPageLoadForUploadComplete(mode))
-            } else {
-              Redirect(controllers.routes.OtherReasonController.onPageLoadForFirstFileUpload(mode))
-            }
-          }
+          val uploadState: Boolean = previousUploadsState.exists(_.count(_.fileStatus == READY) > 0)
+          (mode, uploadState) match {
+            case (NormalMode | CheckMode, true) => Redirect(controllers.routes.OtherReasonController.onPageLoadForUploadComplete(mode))
+            case (NormalMode | CheckMode, false) => Redirect(controllers.routes.OtherReasonController.onPageLoadForFirstFileUpload(mode))
```

Comment:
I'm just thinking, do we now need to match on the `mode`? Could we do something like this:
 Rename uploadState to hasReadyUploads and have an if statement like: 
```
if(hasReadyUploads) Redirect(controllers.routes.OtherReasonController.onPageLoadForUploadComplete(mode))
else Redirect(controllers.routes.OtherReasonController.onPageLoadForFirstFileUpload(mode))
```

Reply:
👍